### PR TITLE
Updated upgrade.sh and readme file for redefined pods

### PR DESCRIPTION
**Pull request by cahuja123**
Signed-off-by: Chirag Ahuja <chirag.ahuja@seagate.com>

# Problem Statement
- Modified upgrade.sh for re-defined Pods

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered test/deploy/kubernetes/README.md](https://github.com/Seagate/cortx-prvsnr/blob/EOS-26196-Modified-upgrade.sh-for-redefined-pods/test/deploy/kubernetes/README.md)

### CODE REVIEW

File: test/deploy/kubernetes/upgrade.sh [hunk 1]
```diff
@@ -112,7 +115,7 @@ while IFS= read -r line; do
             printf "${PASSED}PASSED${NC}\n"
         fi
     fi
-done <<< "$(kubectl get pods --namespace=$namespace | grep 'control-node-.*\|storage-node.-.*')"
+done <<< "$(kubectl get pods --namespace=$namespace | grep 'data-node.-*.\|control-node-.*\|server-node.-.*\|ha-node-.*\|storage-node.-.*')"
```
Comment (tanujashinde0405) [inline test/deploy/kubernetes/upgrade.sh:118]: Pull double quotes for variables-> "$namespace"

File: test/deploy/kubernetes/upgrade.sh [hunk 2]
```diff
@@ -125,23 +128,70 @@ printf "\n\n"
...
+if [ $REDEFPODS = true ]; then
+    # Set new Image for HA POD
+    kubectl set image deployment ha-node cortx-setup=$UPGRADE_IMAGE cortx-ha-fault-tolerance=$UPGRADE_IMAGE cortx-ha-health-monitor=$UPGRADE_IMAGE cortx-ha-k8s-monitor=$UPGRADE_IMAGE;
+
+    # Set new Image for Data POD and Server POD
+    for NODE_INDEX in $(seq 1 $MAXNODES); do
+        DATA_NODE="data-node$NODE_INDEX"
+        SERVER_NODE="server-node$NODE_INDEX"
+        kubectl set image deployment $DATA_NODE cortx-setup=$UPGRADE_IMAGE cortx-motr-hax=$UPGRADE_IMAGE cortx-motr-confd=$UPGRADE_IMAGE cortx-ios1=$UPGRADE_IMAGE cortx-ios2=$UPGRADE_IMAGE;
+        kubectl set image deployment $SERVER_NODE cortx-setup=$UPGRADE_IMAGE cortx-motr-hax=$UPGRADE_IMAGE cortx-s31=$UPGRADE_IMAGE  cortx-s32=$UPGRADE_IMAGE cortx-s3-auth-server=$UPGRADE_IMAGE  cortx-s3-haproxy=$UPGRADE_IMAGE cortx-s3-bgworker=$UPGRADE_IMAGE;
```
Comment (tanujashinde0405) [inline test/deploy/kubernetes/upgrade.sh:186]: put double quotes on $DATA_NODE , $SERVER_NOD, $STOARGE_NODE, $UPGRADE_IMAGE

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by tanujashinde0405 — APPROVED** [review]
LGTM, Please make the upgrade executable.

File: test/deploy/kubernetes/upgrade.sh (hunk 2 above)
Comment (cahuja123) [reply test/deploy/kubernetes/upgrade.sh]: Done

**Comment by cahuja123** [comment]
> LGTM, Please make the upgrade executable.

Done.

**Review by shriya-deshmukh — APPROVED** [review]

**Review by kailashyadav11 — APPROVED** [review]
LGTM

**Pull request merged**